test/hotspot/jtreg/serviceability/jvmti/vthread/FollowReferences/VThreadStackRefTest.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @requires vm.jvmti
+ * @requires vm.continuations
+ * @enablePreview
+ * @run main/othervm/native
+ *      -Djava.util.concurrent.ForkJoinPool.common.parallelism=1
+ *      -agentlib:VThreadStackRefTest
+ *      VThreadStackRefTest
+ */
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.Stream;
+import java.util.concurrent.CountDownLatch;
+
+public class VThreadStackRefTest {
+
+    static volatile boolean timeToStop = false;
+    static int i = -1;
+
+    public static void main(String[] args) throws InterruptedException {
+        CountDownLatch dumpedLatch = new CountDownLatch(1);
+        Thread vthreadMounted = Thread.ofVirtual().start(() -> {
+            Object referenced = new VThreadMountedReferenced();
+            System.out.println(referenced.getClass());
+            while (!timeToStop) {
+                if (++i == 10000) {
+                    i = 0;
+                }
+            }
+            await(dumpedLatch);
+            System.out.println(referenced.getClass());
+        });
+        Thread vthreadUnmounted = Thread.ofVirtual().start(() -> {
+            Object referenced = new VThreadUnmountedReferenced();
+            System.out.println(referenced.getClass());
+            await(dumpedLatch);
+            System.out.println(referenced.getClass());
+        });
+        Thread vthreadJNIUnmounted = Thread.ofVirtual().start(() -> {
+            createObjAndCallback(VThreadUnmountedJNIReferenced.class,
+                new Runnable() {
+                    public void run() {
+                        await(dumpedLatch);
+                    }
+                });
+        });
+
+        Thread vthreadEnded = Thread.ofVirtual().start(() -> {
+            Object referenced = new VThreadUnmountedEnded();
+            System.out.println(referenced.getClass());
+        });
+        Thread pthread = Thread.ofPlatform().start(() -> {
+            Object referenced = new PThreadReferenced();
+            System.out.println(referenced.getClass());
+
+            await(dumpedLatch);
+            System.out.println(referenced.getClass());
+        });
+        vthreadEnded.join();
+
+        Thread.sleep(2000); // wait for reference and unmount
+
+        TestCase[] testCases = new TestCase[] {
+            new TestCase(VThreadMountedReferenced.class, 1, vthreadMounted.getId()),
+            new TestCase(VThreadUnmountedReferenced.class, 1, vthreadUnmounted.getId()),
+            new TestCase(VThreadUnmountedJNIReferenced.class, 1, vthreadJNIUnmounted.getId()),
+            new TestCase(PThreadReferenced.class, 1, pthread.getId()),
+            // expected to be unreported as stack local
+            new TestCase(VThreadUnmountedEnded.class, 0, 0)
+        };
```

Comment:
I think it would be useful the have a test case which has expected_cnt > 1.

Reply:
expected_cnt > 1 means there are references to 2 objects of the class or 2 references to the same object.
I don't see how this would improve test coverage.
1 (or 0) reference to each object helps to keep the test simple